app/validators/candidate_interface/continuous_applications/course_selection_validator.rb

```diff
@@ -4,24 +4,36 @@ module ContinuousApplications
     # Conditional exists to remove the course that is being edited from the checks for duplication
     class CourseSelectionValidator < ActiveModel::Validator
       def validate(record)
-        return unless record.wizard.current_application
+        @record = record
 
-        scope = record.wizard.current_application.application_choices.joins(:course_option)
-        scope = scope.where.not({ course_option: { course_id: record.course.id } }) if editing?(record)
-        exists = scope.exists?(course_option: { course_id: record.course.id })
+        return unless current_application
+
+        scope = current_application.application_choices.joins(:course_option)
+        scope = omit_current_application_choice(scope) if editing?
+        exists = scope
+                  .where({ status: ApplicationStateChange::NON_REAPPLY_STATUSES })
+                  .exists?(course_option: { course_id: record.course.id })
 
         if exists
-          record.errors.add :base, 'You have already applied to this course'
+          @record.errors.add :base, 'You have already applied to this course'
         end
       end
 
       # Only validate against existing application choice that are not being edited
-      def editing?(record)
-        return false unless record.wizard.edit?
+      def editing?
+        return false unless @record.wizard.edit?
+
+        course_id = @record.wizard.application_choice.course.id
 
-        choice_being_edited = record.wizard.application_choice
+        course_id == @record.course_id.to_i
+      end
+
+      def omit_current_application_choice(scope)
+        scope.where.not({ course_option: { course_id: @record.course.id } })
```

Comment:
This is incorrect, this needs to remove the present `ApplicationChoice#id` from consideration when checking for duplicates, not disregard all choices that exist on the application form.